imagersite/imager_profile/views.py

```diff
@@ -0,0 +1,59 @@
+from django.shortcuts import render
+from django.http import HttpResponse
+from imager_images.models import Photos, Albums
+from django.core.urlresolvers import reverse_lazy
+from django.conf import settings
+from django.contrib.auth.models import User
+from imager_profile.models import ImagerProfile
+
+
+def home_view(request):
+    """Home view callable, for the home page."""
+    import random
+    photos = Photos.objects.all()
+    if len(photos) > 0:
+        photo_url = random.choice(photos).image.url
+    else:
+        photo_url = settings.media_url + "standard.jpg"
+    return render(request, "imagersite/home.html", {"photo_url": photo_url})
+
+
+def user_profile_view(request):
+    """Profile view callable for user's personal profile."""
+    if request.user.is_authenticated():
+        user = request.user
+        photos = Photos.objects.all().filter(photographer_id=user.profile.id)
+        public_photos = photos.filter(published='PU').count()
+        private_photos = photos.filter(published="PR").count()
+        shared_photos = photos.filter(published="SH").count()
+        albums = Albums.objects.all().filter(id=user.profile.id)
+        return render(
+            request, "imager_profile/profile.html", {
+                                        "user": user,
+                                        "private_photos": private_photos,
+                                        "public_photos": public_photos,
+                                        "shared_photos": shared_photos,
+                                        "albums": albums})
+
+
+def profile_view(request, username):
+    """Profile view callable for all profiles."""
+    user = ImagerProfile.objects.get(user__username=username).user
+    # import pdb; pdb.set_trace()
+    photos = Photos.objects.all().filter(id=user.id)
```

Comment:
This isn't getting the photos associated with the user. It's getting the photo where its ID is equal to whatever the user's ID is.